esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -371,12 +372,71 @@ def compute_overall_mean(dataset: 'xr.Dataset',
     return overall_mean
 
 
+def combine_dimension(dataset: 'xr.DataArray', dimn: str) -> 'xr.DataArray':
+    """Combine ensemble members of the same model along the given dimension."""
+    model_ensemble = dataset[dimn].values
+    models = [me.split('_')[0] for me in model_ensemble]
+    dataset['model'] = xr.DataArray(models, dims=dimn)
```

Comment:
I wonder if it makes sense to groupby both dimensions simulatenously, something like:
```suggestion
    if 'perfect_model_ensemble' in dataset.dims:
        models = np.meshgrid(models, models)
        model = xr.DataArray(models, dims=[both dims])
    else:
        model = xr.DataArray(models, dims=[single dim]
````
```

Reply:
Also see my comment below

Reply:
I like the idea but I could not get it to work and I'm not sure if it can work. If I group both dimensions at the same time I end up with a one-dimensional array, but I will want a matrix... Still simplified a lot of this function though (see other comments)